src/base/indexrange.h

```diff
@@ -123,8 +166,8 @@ class IndexRange
     }
 
 private:
-    IndexType m_first;
-    IndexDiffType m_size;
+    const IndexType m_first;
```

Comment:
I don't think these two const are necessary. It makes the class non-assignable which has no point, IMO.